### PR TITLE
 Adapt LLDB to support location-less Swift debug info for zero-sized t…

### PR DESCRIPTION
**Pull request by adrian-prantl**
…ypes.

rdar://problem/51343998

### CODE REVIEW

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/25435
@swift-ci test

File: source/Core/ValueObjectVariable.cpp [hunk 1]
```diff
@@ -139,18 +151,8 @@ bool ValueObjectVariable::UpdateValue() {
...
-        llvm::Optional<uint64_t> size =
-            var_type.GetByteSize(exe_ctx.GetBestExecutionContextScope());
-        if (size && *size == 0)
-          m_value.SetCompilerType(var_type);
-        else
-          m_error.SetErrorString("empty constant data");
-      }
-    }
+    } else
+      m_error.SetErrorString("empty constant data");
```
Comment (adrian-prantl) [inline source/Core/ValueObjectVariable.cpp:155]: Note that the moved code already was also swift-lldb-specific.

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift/pull/25435
@swift-ci test

File: source/Symbol/SwiftASTContext.cpp [hunk 2]
```diff
@@ -5964,9 +5964,7 @@ SwiftASTContext::GetBitSize(lldb::opaque_compiler_type_t type,
   swift::CanType swift_can_type(GetCanonicalSwiftType(type));
   if (swift_can_type->hasTypeParameter()) {
     if (!exe_scope)
-      // FIMXE: This should be {}. lldb::Type also needs to also take an
-      //        optional size and resolve it at runtime if it is None.
-      return GetPointerByteSize() * 8;
+      return {};
```
Comment (dcci) [inline source/Symbol/SwiftASTContext.cpp:5967]: Nice!

File: source/Core/ValueObjectVariable.cpp [hunk 3]
```diff
@@ -139,18 +151,8 @@ bool ValueObjectVariable::UpdateValue() {
       if (m_data.GetDataStart() && m_data.GetByteSize())
         m_value.SetBytes(m_data.GetDataStart(), m_data.GetByteSize());
       m_value.SetContext(Value::eContextTypeVariable, variable);
-    } else {
```
Comment (dcci) [inline source/Core/ValueObjectVariable.cpp:142]: Why did you need to move this code?

File: source/Core/ValueObjectVariable.cpp [hunk 4]
```diff
@@ -130,6 +130,18 @@ bool ValueObjectVariable::UpdateValue() {
...
   Variable *variable = m_variable_sp.get();
+  // Check if the type has size 0. If so, there is nothing to update.
+  CompilerType var_type(GetCompilerTypeImpl());
+  if (var_type.IsValid()) {
+    ExecutionContext exe_ctx(GetExecutionContextRef());
+    llvm::Optional<uint64_t> size =
+      var_type.GetByteSize(exe_ctx.GetBestExecutionContextScope());
+    if (size && *size == 0) {
+      m_value.SetCompilerType(var_type);
+      return m_error.Success();
```
Comment (adrian-prantl) [inline source/Core/ValueObjectVariable.cpp:141]: Because of this new early exit.

Comment (dcci) [reply source/Core/ValueObjectVariable.cpp:141]: Oh, I see now.

**Review by dcci — APPROVED** [review]
